### ISSUE TITLE
Ubuntu Dependency Script is throwing repository errors, no longer runs successfully

### ISSUE THREAD

**Issue by nikito** [toverainc/willow-inference-server#91]
I was reinstalling my WIS after cleaning up my environment, and when I went to run the /deps/ubuntu.sh script I now get errors about the libnvidia-container repository "no longer has a Release file". And an error code of 404. Not sure if a recent change was made in their repository that broke the script?
Terminal messages below:
nbento@docker-server-1:/willow-inference-server$ ./deps/ubuntu.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 21926  100 21926    0     0   195k      0 --:--:-- --:--:-- --:--:--  196k
# Executing docker install script, commit: c2de0811708b6d9015ed1a2c80f02c9b70c8ce7b
Warning: the "docker" command appears to already exist on this system.

If you already have Docker installed, this script can cause trouble, which is
why we're displaying this warning and provide the opportunity to cancel the
installation.

If you installed the current Docker package using this script and are using it
again to update Docker, you can safely ignore this message.

You may press Ctrl+C now to abort this script.
+ sleep 20
+ sudo -E sh -c apt-get update -qq >/dev/null
[sudo] password for nbento:
E: The repository 'https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  Release' no longer has a Release file.
File '/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg' exists. Overwrite? (y/N) curl: (22) The requested URL returned error: 404
                                                                                                                                         y
gpg: no valid OpenPGP data found.
Hit:1 http://us.archive.ubuntu.com/ubuntu jammy InRelease
Ign:2 https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  InRelease
Hit:3 http://us.archive.ubuntu.com/ubuntu jammy-updates InRelease
Hit:4 https://download.docker.com/linux/ubuntu jammy InRelease
Err:5 https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  Release
  404  Not Found [IP: 185.199.110.153 443]
Hit:6 http://us.archive.ubuntu.com/ubuntu jammy-backports InRelease
Hit:7 http://us.archive.ubuntu.com/ubuntu jammy-security InRelease
Reading package lists... Done
E: The repository 'https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/amd64  Release' no longer has a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

**Comment by kristiankielhofner** [comment]
Generally that script is to be used on first time installs only. Looking things over it shouldn't break anything but I haven't been able to test what happens if you run it in an existing install.

That said I've tested a bunch of upgrades, installing from scratch, etc without issue.

All I can think is that re-running it broke the container repos (or you had some temporary network issue, possibly). I have:

```deb [signed-by=/usr/share/keyrings/nvidia-container-toolkit-keyring.gpg] https://nvidia.github.io/libnvidia-container/stable/ubuntu18.04/$(ARCH) /```

Can you ```cat /etc/apt/sources.list.d/nvidia-container-toolkit.list``` and compare?

**Comment by nikito** [comment]
Yeah figured it probably only needed to be ran once, was just running through each step of the readme to be safe 😆 
Everything did install fine and is working correctly, so no issues there.

Checked my nvidia-container-toolkit.list and have the same entry as you, so probably just something weird with the container repo or something like that.

**Comment by kristiankielhofner** [comment]
Great! I updated the README in the wisng branch to (hopefully) make the instructions for deps and install clearer.
